crates/red_knot_python_semantic/src/semantic_model.rs

```diff
@@ -205,7 +205,7 @@ mod tests {
         let model = SemanticModel::new(&db, foo);
         let ty = function.ty(&model);
 
-        assert!(matches!(ty, Type::Function(_)));
+        assert!(matches!(ty, Type::FunctionLiteral(_)));
```

Comment:
nit: we could add `is_function_literal()` and `is_class_literal` methods to `Type`, and use them here

Reply:
This method does not exist :smile:. I could do `assert!(ty.into_function_literal_type().is_some())`, but I don't think that's much better than matching on the variant?

Reply:
> This method does not exist 😄

I know! Sorry for being unclear. I'm suggesting we could add `Type::is_function_literal()` and `Type::is_class_literal()` methods to the `Type` enum, for consistency with our existing `Type::is_unbound` and `Type::is_never` methods. It doesn't _have_ to be done in this PR, but it also _might as well_ if we're touching these lines anyway ;)

Reply:
No, sorry. I misread.

`is_function_literal` does not exist, but `is_class_literal` exists. It does something different from just matching against `Type::ClassLiteral` though :see_no_evil: 

```rs
pub fn is_class_literal(&self, db: &'db dyn Db) -> bool {
    match self {
        Type::Union(union) => union.elements(db).iter().all(|ty| ty.is_class_literal(db)),
        Type::ClassLiteral(_) => true,
        // / TODO include type[X], once we add that type
        _ => false,
    }
}
```

So if we want to do this properly, I guess we would have to rename the existing `is_class_literal` function...

Reply:
> So if we want to do this properly, I guess we would have to rename the existing `is_class_literal` function...

We're well down the renaming rabbit hole now. May as well 😆

It seems like the question the existing `is_class_literal` method is _actually_ trying to answer is "Is the type a subtype of `Instance(builtins.type)`?". Maybe we could try to just get rid of it and make `is_subtype_of` work correctly for `Instance(builtins.type)`, so that it recognises all `Type::ClassLiteral`s as subtypes

Reply:
Did that now.